notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "train_data.describe(percentiles=[.25, 0.375, .5, .75, .875]).drop([\"count\"])       \n",
+    "\n",
+    "# I dropped \"count\" row to get rid of exponentioal number presentation\n",
+    "# And added .875 percentile to understand approximately how many objects have no onpromotion"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "6yDCEDYYGveo",
+    "outputId": "839affcc-2928-4892-c798-459ff56b59f0"
+   },
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>The only categorical feature (in this table) \"family\" has 33 possible unique values (not so a lot) <br>\n",
+    "That means we can easily use one-hot-encoding during model training</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "family_values = train_data[\"family\"].unique()\n",
+    "family_values"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 40))\n",
+    "for i, value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(len(family_values)//3, 3, i+1)\n",
+    "  current_value_data = train_data[train_data[\"family\"] == value].groupby([\"date\"]).mean()\n",
+    "  plt.xlim=(0, 1750)\n",
+    "  plt.ylim=(0, current_value_data[\"sales\"].max())\n",
+    "  plt.scatter(x=np.arange(len(current_value_data.index)), y=current_value_data[\"sales\"])\n",
+    "  plt.title(value)\n",
+    "plt.xlim=(0, 1750)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>Actually, these plots don't really help us, because, looking on them, we can only catch some common \"trends\", which describe global changes in some products sales.<br>For example, we can be sure that on the 1th January every year number of sales is equal to 0 (actually, it is necessary to check, but I think it is obvious). <br> Also we can say that the number of sales in general has a positive dynamic (for example, 'automotive', 'bread/bakery', 'grocery i', 'personal care', ...).<br>Some of the product types have a negative dynamic (such as lingerie).<br>Some goods have very interesting sales distribution (books, produce, froxen foods, ladieswear, ...), and we can't say right now, what is the reason of that.<br>Also some goods families have seasonal increase in sales('school and office supplies', 'liquor, wine, beer', 'grocery ii', 'frozen foods'). </i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 110))\n",
```

Comment:
Probably your previous plot is excess, since this plot show almost the same + trend.

My only question here ralates to these categories and the points highlighted: what happened there?
![image](https://user-images.githubusercontent.com/24436463/179365056-1fda41fe-1462-4f29-af0a-46304a6b508f.png)

Probably it worth using rolling averages or exlcude these outliers to take a closer look at trends in these cases.